### DISCUSSION TITLE
Hello. I am encountering a linking error with ktx.lib

### DISCUSSION THREAD

**Discussion by smikim** [KhronosGroup/KTX-Software#943] [Q&A] [answered]
<img width="1226" alt="ktxliberror" src="https://github.com/user-attachments/assets/11ffce99-6dbb-4e2b-9321-e3a596353cd7">

Hello. I am encountering a linking error with ktx.lib. I am using the Windows operating system, and I followed these steps:

git clone https://github.com/KhronosGroup/KTX-Software.git
cd KTX-Software
git lfs install
git lfs checkout
cmake .. -G "Visual Studio 17 2022" -A x64
cmake --build . --config Release

After that, I copied the ktx.exp, ktx.lib, ktx.exe, and ktx.dll files from the build/release directory. I placed the ktx.dll file in the same directory as my program's executable file. In Visual Studio, I added ktx.lib to the "Additional Dependencies" under "Linker -> Input" in the project properties. I also added the path containing ktx.exp, ktx.lib, and ktx.exe to the "Additional Library Directories" under "Linker -> General".
Despite these steps, I am still encountering a linking error. What should I do? If there are any additional details needed, please let me know.

**Comment by MarkCallow** [comment, marked as answer]
Sorry for failing to answer this for so long. From the symbol names in the error messages it looks like your application has been built to link with a static ktx.lib but the ktx.lib built with the default cmake settings is companion to a dll so the symbol names have a `_imp` suffix.
